sphinx/directives/__init__.py

```diff
@@ -296,7 +296,7 @@ def run(self) -> list[Node]:
             # If ``:no-index:`` is set, or there are no ids on the node
             # or any of its children, then just return the index node,
             # as Docutils expects a target node to have at least one id.
-            if node_ids := [node_id for el in node.findall(nodes.Element)
+            if node_ids := [node_id for el in node.findall(nodes.Element)  # type: ignore[var-annotated]
```

Comment:
these are the overloads in typeshed:

```python
    @overload
    def findall(
        self, condition: type[_N], include_self: bool = True, descend: bool = True, siblings: bool = False, ascend: bool = False
    ) -> Generator[_N, None, None]: ...
    @overload
    def findall(
        self,
        condition: Callable[[Node], bool] | None = None,
        include_self: bool = True,
        descend: bool = True,
        siblings: bool = False,
        ascend: bool = False,
    ) -> Generator[Node, None, None]: ...
```

so that overload does exist, but mypy is still not able to infer.
it's an ugly bit of code